python-package/xgboost/spark/core.py

```diff
@@ -267,6 +267,18 @@ def _validate_params(self):
                 "If features_cols param set, then features_col param is ignored."
             )
 
+        if self.getOrDefault(self.objective) is not None:
```

Comment:
It seems not make sense (to me) to check the type of a parameter that already has the typeConverter

Reply:
The "objective" param is automatically generated, currently it does not have type converter.

Reply:
I still hope we can find a common solution for the type converter in the following PR